cypress/integration/rendering/mindmap.spec.js

```diff
@@ -1,12 +1,32 @@
 import { imgSnapshotTest, renderGraph } from '../../helpers/util.js';
 
+/**
+ * Check whether the SVG Element has a Mindmap root
+ *
+ * Sometimes, Cypress takes a snapshot before the mermaid mindmap has finished
+ * generating the SVG.
+ *
+ * @param {JQuery<SVGSVGElement>} $p - The element to check.
```

Comment:
Cypress injects it.

I've switched this file to TypeScript to hide the JSDoc ESLint warning, and added a `cypress/tsconfig.json` file that declares the `cypress` types (`JQuery` is part of that).

I also encountered some `mindmap` tests in `mermaid.spec.js`, so I've moved those in `mindmap.spec.js` as well, because previously sometimes made a snapshot before rendering was complete!